src/cargo/ops/cargo_doc.rs

```diff
@@ -44,13 +44,15 @@ pub fn doc(manifest_path: &Path,
     try!(ops::compile(manifest_path, &options.compile_opts));
 
     if options.open_result {
-        let name = match options.compile_opts.spec {
-            Some(spec) => try!(PackageIdSpec::parse(spec)).name().replace("-", "_").to_string(),
-            None => {
-                match lib_names.iter().chain(bin_names.iter()).nth(0) {
-                    Some(s) => s.to_string(),
-                    None => return Ok(())
-                }
+        let name = if options.compile_opts.spec.len() > 0{
+            // TODO
```

Comment:
Could you have this generate an error if `--open` is passed and multiple packages are specified?